flake: metrics cache tests

This happens frequently, see [run](https://github.com/coder/coder/actions/runs/3285969221/jobs/5413582505)
```
=== FAIL: coderd/metricscache TestCache_BuildTime/one/start (0.51s)
    metricscache_test.go:333: 
        	Error Trace:	/Users/runner/work/coder/coder/coderd/metricscache/metricscache_test.go:333
        	            				/Users/runner/work/coder/coder/coderd/metricscache/asm_amd64.s:1594
        	Error:      	Should NOT be empty, but was {<nil> <nil> <nil>}
        	Test:       	TestCache_BuildTime/one/start
    --- FAIL: TestCache_BuildTime/one/start (0.51s)

=== FAIL: coderd/metricscache TestCache_BuildTime/two/stop (0.51s)
    metricscache_test.go:333: 
        	Error Trace:	/Users/runner/work/coder/coder/coderd/metricscache/metricscache_test.go:333
        	            				/Users/runner/work/coder/coder/coderd/metricscache/asm_amd64.s:1594
        	Error:      	Should NOT be empty, but was {<nil> <nil> <nil>}
        	Test:       	TestCache_BuildTime/two/stop
    --- FAIL: TestCache_BuildTime/two/stop (0.51s)

=== FAIL: coderd/metricscache TestCache_BuildTime/three/delete (0.51s)
    metricscache_test.go:333: 
        	Error Trace:	/Users/runner/work/coder/coder/coderd/metricscache/metricscache_test.go:333
        	            				/Users/runner/work/coder/coder/coderd/metricscache/asm_amd64.s:1594
        	Error:      	Should NOT be empty, but was {<nil> <nil> <nil>}
        	Test:       	TestCache_BuildTime/three/delete
    --- FAIL: TestCache_BuildTime/three/delete (0.51s)
```
cc @ammario

--- a/coderd/metricscache/metricscache_test.go
+++ b/coderd/metricscache/metricscache_test.go
@@ -330,7 +330,7 @@ func TestCache_BuildTime(t *testing.T) {
 			if tt.want.loads {
 				require.Eventuallyf(t, func() bool {
 					stats := cache.TemplateBuildTimeStats(template.ID)
-					return assert.NotEmpty(t, stats)
+					return stats != codersdk.TemplateBuildTimeStats{}
 				}, testutil.WaitLong, testutil.IntervalMedium,
 					"BuildTime never populated",
 				)

